src/odemis/gui/comp/stream.py

```diff
@@ -1582,3 +1582,534 @@ def _set_warning(self):
+        """
+        calibration_ctrl = ComboBox(self, value=choices[0], choices=choices, size=(100, -1),
+                                    style=wx.CB_READONLY | wx.BORDER_NONE)
+        self._add_ctrl(calibration_ctrl)
+        return calibration_ctrl
+
+    def _add_ctrl(self, ctrl, stretch=False):
+        """ Add the given control to the panel
+
+        :param ctrl: (wx.Control) Control to add to the header panel
+        :param stretch: True if the control should expand to fill space
+        """
+        self._panel_sizer.Add(
+            ctrl,
+            proportion=1 if stretch else 0,
+            flag=(wx.RIGHT | wx.ALIGN_CENTRE_VERTICAL | wx.RESERVE_SPACE_EVEN_IF_HIDDEN),
+            border=self.BUTTON_BORDER_SIZE
+        )
+
+    def _on_size(self, event):
+        """ Handle the wx.EVT_SIZE event for the Expander class """
+        self.SetSize((self._parent.GetSize().x, -1))
+        self.Layout()
+        self.Refresh()
+        event.Skip()
+
+
+class FastEMCalibrationBar(wx.Panel):
+    """
+    The whole panel containing the panel with the calibration buttons.
+    """
+
+    DEFAULT_BORDER = 2
+    DEFAULT_STYLE = wx.BOTTOM | wx.EXPAND
+
+    def __init__(self, *args, **kwargs):
+        kwargs.pop('add_button', False)  # remove add_button kwarg
+        wx.Panel.__init__(self, *args, **kwargs)
+
+        self._sz = wx.BoxSizer(wx.VERTICAL)
+        self.SetSizer(self._sz)
+
+    def add_calibration_panel(self, panel):
+        """
+        This method adds the calibration panel to the calibration bar. Should only be called once.
+        panel (FastEMCalibrationPanel): the calibration panel
+        """
+        self._sz.Insert(0, panel, flag=self.DEFAULT_STYLE, border=self.DEFAULT_BORDER)
+        panel.Layout()
+
+
+class FastEMCalibrationPanel(wx.Panel):
+    """
+    Panel for the calibration buttons.
+    """
+
+    def __init__(self, parent, layout,
+                 wid=wx.ID_ANY, pos=wx.DefaultPosition, size=wx.DefaultSize,
+                 style=wx.CP_DEFAULT_STYLE, name="CalibrationPanel"):
+        """
+        layout (list of lists of int): layout of scintillator grid, given as 2D list of scintillator positions,
+        e.g. [[6, 5, 4], [3, 2, 1]]
+        """
+        assert(isinstance(parent, FastEMCalibrationBar))
+        wx.Panel.__init__(self, parent, wid, pos, size, style, name)
+
+        self.buttons = {}  # int --> wx.Button
+
+        self._panel_sizer = wx.BoxSizer(wx.VERTICAL)
+        self.SetSizer(self._panel_sizer)
+
+        # Calibration Grid
+        nrows = len(layout)
+        ncols = max([len(row) for row in layout])
+        calgrid_sz = wx.GridBagSizer(nrows, ncols)
+        for row_idx, row in enumerate(layout):
+            for col_idx, elem in enumerate(row):
+                subsz = wx.BoxSizer(wx.HORIZONTAL)
+                btn = wx.Button(self, wx.ALL | wx.ALIGN_CENTER, label="?", size=(30, 30))
+                btn.SetBackgroundColour("#999999")
```

Comment:
It seems this colour keeps being reused. I think we should add this colour to the constants in odemis.gui. Or, alternatively, use one from there. It seems it could be called something like FG_COLOUR_....